src/libs/actions/Wallet.ts

```diff
@@ -0,0 +1,386 @@
+import Onyx, {OnyxUpdate} from 'react-native-onyx';
+import {ValueOf} from 'type-fest';
+import * as API from '@libs/API';
+import {PrivatePersonalDetails} from '@libs/GetPhysicalCardUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import {WalletAdditionalQuestionDetails} from '@src/types/onyx';
+import * as OnyxCommon from '@src/types/onyx/OnyxCommon';
+
+type WalletTerms = {
+    hasAcceptedTerms: boolean;
+    reportID: string;
+};
+
+type WalletQuestionAnswer = {
+    question: string;
+    answer: string;
+};
+
+type IdentityVerification = {
+    onfidoData: string;
+};
+
+type PersonalDetails = {
+    phoneNumber: string;
+    legalFirstName: string;
+    legalLastName: string;
+    addressStreet: string;
+    addressCity: string;
+    addressState: string;
+    addressZip: string;
+    dob: string;
+    ssn: string;
+};
+
+/**
+ * Fetch and save locally the Onfido SDK token and applicantID
+ * - The sdkToken is used to initialize the Onfido SDK client
+ * - The applicantID is combined with the data returned from the Onfido SDK as we need both to create an
+ *   identity check. Note: This happens in Web-Secure when we call Activate_Wallet during the OnfidoStep.
+ */
+function openOnfidoFlow() {
+    const optimisticData: OnyxUpdate[] = [
+        {
+            // Use Onyx.set() since we are resetting the Onfido flow completely.
+            onyxMethod: Onyx.METHOD.SET,
+            key: ONYXKEYS.WALLET_ONFIDO,
+            value: {
+                isLoading: true,
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.WALLET_ONFIDO,
+            value: {
+                isLoading: false,
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.WALLET_ONFIDO,
+            value: {
+                isLoading: false,
+            },
+        },
+    ];
```

Comment:
This is a new convention, but since the the `successData` and `failureData` are the same, you can use a single object named `finallyData` here instead.

Reply:
Same for any others in this file 🙇

Reply:
@NikkiWines Done!

Reply:
Thanks for updating this. We're looking into adding `finallyData` in the network layer [instead](https://github.com/Expensify/App/pull/33530/) so you can actually revert these changes for now (sorry, should've held off on suggesting we use `finallyData`) or we can wait until the corresponding PR is merged and update this one accordingly.

Reply:
> we can wait until the corresponding PR is merged and update this one accordingly

Let's wait then! 😄 @NikkiWines @BartoszGrajdek